addons/mail/static/src/utils/common/misc.js

```diff
@@ -69,3 +69,30 @@ export function onChange(target, key, callback) {
 export function closeStream(stream) {
     stream?.getTracks?.().forEach((track) => track.stop());
 }
+
+/**
+ * Compare two objects by Luxon datetime. If both objects have the same datetime
+ * or if both objects don't have a datetime, fallback is returned.
+ *
+ * @param {object} o1
+ * @param {object} o2
+ * @param {string} datetimeKey
+ * @param {object} param4
+ * @param {boolean} [param4.desc]
+ * @param {any} [param4.fallback]
+ * @returns {number} Negative if o1 is before o2, positive if o1 is after o2.
+ */
+export function compareDatetime(o1, o2, datetimeKey, fallbackValue, { desc, fallback } = {}) {
+    const d1 = desc ? o1[datetimeKey] : o2[datetimeKey];
+    const d2 = desc ? o2[datetimeKey] : o1[datetimeKey];
+    if (fallback && ((!d1 && !d2) || d1.equals(d2))) {
+        return fallback;
+    }
```

Comment:
I'm not fan of the fallback mechanism being built-in, I think the method should just return 0 in that case and the fallback handled outside.

Reply:
The actual fallback we want is to have a "compare list", and providing to this list various methods, going to the next step while it is equal.